### PR TITLE
Update CoMID relations

### PR DESCRIPTION
**Pull request by thomas-fossati**
This PR updates the list of CoMID relations as per discussion captured in #20.

It also cleans up the CDDL a bit.

### CODE REVIEW

**Comment by thomas-fossati** [comment]
(This branch is on top of #33)

File: cddl/concise-mid-tag.cddl [hunk 1]
```diff
@@ -1,62 +1,123 @@
...
+; fashion.
+comid-updates = 3
+; The current tag contains corrections to a previously issued tag. This tag
+; replaces the target tag. The tag-id remains unchanged. The tag-version is
+; incremented.
+comid-replaces = 4
+; The current module contains claims that partially replace the target module
+; claims. This is applied when the module is updated or changes in some
+; fashion.  The target module contains both valid and replaced claims.
+comid-patches = 5
```
Comment (nedmsmith) [inline cddl/concise-mid-tag.cddl:118]: Do we want to define 'patches' yet? Hoping to avoid this complexity if possible.

Comment (thomas-fossati) [reply cddl/concise-mid-tag.cddl:118]: It is the tool we were planning to use in order to model FW patch releases.

File: cddl/concise-mid-tag.cddl [hunk 2]
```diff
@@ -1,62 +1,123 @@
...
+  ? 3 => module-vendor,
+  * $$module-metadata-extension
+}
+
+module-name = text
+
+$module-type /= tagged-oid
+$module-type /= tagged-uuid
+$module-type /= tagged-impl-id
+$module-type /= tagged-ueid
```
Comment (thomas-fossati) [inline cddl/concise-mid-tag.cddl:39]:
```suggestion
```

**Pull request merged**